jax/_src/interpreters/batching.py

```diff
@@ -966,7 +966,10 @@ def _matchaxis_symbolic_zeros(axis_name, sz, name, src, dst, x, sum_match=False)
 
 ### utilities for defining primitives' batching rules
 
-BatchingRule = Callable[..., tuple[Any, Union[None, int, tuple[Union[None, int], ...]]]]
+BatchingRule = Callable[
+    ...,
+    tuple[Any, Union[int, None, tuple[Union[int, None], ...]]]
```

Comment:
Here and a few other places: why does it still use `Union` rather than `|`?

Reply:
That's because type aliases are still evaluated eagerly. I was hoping we could "delay" them via

```
T: TypeAlias = 'Foo | Bar'
```

but that feature is only available on 3.10+.

So our only option on 3.9 is to keep using `Union` for type aliases and in `cast(...)`.